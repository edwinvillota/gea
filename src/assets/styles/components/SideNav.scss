@import '../themes/default.scss';

::-webkit-scrollbar {display: none;}

.main__container {
    display: flex;
    justify-content: flex-end;
    width: 100%;
    min-height: 100vh;
    background-color: white;

    .hamburguer__control--wrapper {
        position: fixed;
        display: flex;
        left: 0;
        top: 0;
        height: 40px;
        width: 40px;
        transition: all .2s ease-in;
        background-color: $Accent;
        

        #h__check {
            display: none;
        }

        .hamburguer__label {
            display: flex;
            justify-content: center;
            align-items: center;
            width: 100%;
            height: 100%;
            font-size: 30px;
            cursor: pointer;
        }

        .hamburguer__icon {
            color: white;
        }
    }

    .open {
        left: 300px;
    }

    .sidenav__main--wrapper {
        display: flex;
        flex-direction: column;
        background-color: $Dark;
        position: fixed;
        height: 100%;
        max-height: 100vh;
        transition: all .2s ease-in;
        min-width: 300px;


        .user__section {
            display: flex;
            flex-direction: column;
            align-items: center;
            background-color: $Dark_gray;
            height: 260px;
            padding: 1em 0;

            .user__avatar {
                width: 135px;
                height: 135px;
                border-radius: 100%;
                border: solid 3px $Accent;
                background-color: $Mid_gray;
            }

            .user__name {
                font-family: $Standar;
                font-size: 20px;
                color: $Light_gray;
                font-weight: lighter;
            }

            .user__jobtitle {
                margin-top: -17px;
                padding: 0;
                font-family: $Standar;
                color: $Accent;
                font-size: 16px;
                font-weight: lighter;
            }

            .user__rol {
                margin-top: -12px;
                font-family: $Standar;
                font-weight: lighter;
                font-size: 12px;
                color: $Mid_gray;
            }
        }

        .modules__section {
            display: flex;
            flex-direction: column;
            max-height: calc(100vh - 260px);
            overflow-x: visible;
            overflow-y: scroll;
        }
    }

    .show {
        left: 0;
    }

    .hide {
        left: -300px;
    }

    .main__content {
        display: flex;
        padding: 0 2em;
        width: calc(100% - 340px);
        transition: all .2s ease-in;
    }

    .content__expanded {
        width: calc(100% - 50px);
    }

}
