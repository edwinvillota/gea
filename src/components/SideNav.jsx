import React, { useState } from 'react'
import { IoMdMenu, IoMdClose } from 'react-icons/io'
import classNames from 'classnames'

// Assets
import '../assets/styles/components/SideNav.scss'

// Components
import SideNavModule from './SideNavModule'

const SideNav = ({children}) => {

    const [visible, setVisibility] = useState(false)

    const handleChangueVisibility = () => {
        console.log(visible)
        setVisibility(!visible)
    }

    const modules = [
        {
            hasSubroutes: false,
            name: 'Home',
            route: '/home',
            icon: 'AiOutlineHome'  
        },
        {
            hasSubroutes: false,
            name: 'Perfil de Usuario',
            route: '/user_profile',
            icon: 'FiUser'
        },
        {
            hasSubroutes: true,
            name: 'Administración',
            route: '/management',
            icon: 'AiOutlineApartment',
            subroutes: [
                {
                    name: 'Usuarios',
                    route: '/management/users',
                    icon: 'FiUsers'
                },
                {
                    name: 'Cargos',
                    route: '/management/job_titles',
                    icon: 'MdWork'
                },
                {
                    name: 'Roles',
                    route: '/management/roles',
                    icon: 'AiOutlineSolution'
                }
            ]
        },
        {
            hasSubroutes: true,
            name: 'Módulos',
            route: '/modules',
            icon: 'AiOutlineApi',
            subroutes: [
                {
                    name: 'Agregar Módulo',
                    route: '/modules/add',
                    icon: 'IoMdAdd'
                },
                {
                    name: 'Eliminar Módulo',
                    route: '/modules/del',
                    icon: 'IoMdRemove'
                }
             ]
        },
        {
            hasSubroutes: true,
            name: 'Control Operativo',
            route: '/operative_control',
            icon: 'AiOutlineLineChart',
            subroutes: [
                {
                    name: 'Contratos',
                    route: '/operative_control/contracts',
                    icon: 'TiDocumentText'
                },
                {
                    name: 'Personal',
                    route: '/operative_control/staff',
                    icon: 'AiOutlineTeam'
                },
                {
                    name: 'Actividades',
                    route: '/operative_control/activities',
                    icon: 'GoTasklist'
                },
                {
                    name: 'Estado de Ejecución',
                    route: '/operative_control/status',
                    icon: 'AiOutlinePieChart'
                }
             ]
        },
        {
            hasSubroutes: true,
            name: 'Inventarios',
            route: '/inventary',
            icon: 'MdStorage',
            subroutes: [
                {
                    name: 'Bodegas',
                    route: '/inventary/warehoses',
                    icon: 'FaWarehouse'
                },
                {
                    name: 'Entradas',
                    route: '/inventary/inputs',
                    icon: 'AiOutlineImport'
                },
                {
                    name: 'Traspasos',
                    route: '/inventary/exchange',
                    icon: 'FaExchangeAlt'
                }
            ]
        },
        {
            hasSubroutes: true,
            name: 'Certificados',
            route: '/certificates',
            icon: 'AiOutlineFileProtect',
            subroutes: [
                {
                    name: 'Cargar desde archivo',
                    route: '/certificates/loadcsv',
                    icon: 'FaFileCsv'
                },
                {
                    name: 'Reconocimiento Optico',
                    route: '/certificate/recognition',
                    icon: 'GiCyberEye'
                }
            ]
        },
        {
            hasSubroutes: false,
            name: 'Cerrar Sesión',
            route: '/logout',
            icon: 'AiOutlineLogout'
        }
    ]

    return (
        <div className='main__container'>
            <div className={classNames("hamburguer__control--wrapper", {'open': visible})}>
                <label htmlFor="h__check" className='hamburguer__label'>
                    {
                        visible
                        ? (<IoMdClose className='hamburguer__icon'/>)
                        : (<IoMdMenu  className='hamburguer__icon'/>)
                    }
                </label>
                <input 
                    type="checkbox" 
                    name="hamburguer" 
                    id="h__check"
                    onClick={
                        handleChangueVisibility
                    }
                    />
            </div>
            <nav className={classNames("sidenav__main--wrapper", {'hide' : !visible, 'show' : visible})}>

                <div className="user__section">
                    <img src="" alt="" className="user__avatar"/>
                    <h1 className="user__name">Edwin Jair Villota Calderon</h1>
                    <h2 className="user__jobtitle">Asistente Administrativo</h2>
                    <h3 className="user__rol">Webmaster</h3>
                </div>

                <div className="modules__section">
                    {
                        modules.map((m, i) => (
                            <SideNavModule 
                                key={i}
                                module = {m}
                                />
                        ))
                    }
                </div>
            </nav>
            <div className={classNames('main__content', {'content__expanded': !visible}) }>
                {children[0]}
            </div>
        </div>
    )
}

export default SideNav
