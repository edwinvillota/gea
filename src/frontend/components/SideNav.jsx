import React, { useState } from 'react';
import { connect } from 'react-redux';
import { IoMdMenu, IoMdClose } from 'react-icons/io';
import classNames from 'classnames';

// Assets
import '../assets/styles/components/SideNav.scss';

// Actions
import {
  logoutRequest,
} from '../actions/authActions';

// Components
import SideNavModule from './SideNavModule';

const SideNav = (props) => {

  const { children } = props;

  const [visible, setVisibility] = useState(false);

  const { user } = props;
  const { modules } = user;

  const handleLogout = () => {
    document.cookie = 'email=';
    document.cookie = 'id=';
    document.cookie = 'username=';
    document.cookie = 'rol_id=';
    document.cookie = 'token=';
    document.cookie = 'scopes=';
    props.logoutRequest();
    window.location.href = '/';
  };

  const handleChangueVisibility = () => {
    setVisibility(!visible);
  };

  return (
    <div className='main__container'>
      <div className={classNames('hamburguer__control--wrapper', { 'open': visible })}>
        <label htmlFor='h__check' className='hamburguer__label'>
          {
            visible ?
              (<IoMdClose className='hamburguer__icon' />) :
              (<IoMdMenu className='hamburguer__icon' />)
          }
        </label>
        <input
          type='checkbox'
          name='hamburguer'
          id='h__check'
          onClick={
            handleChangueVisibility
          }
        />
      </div>
      <nav className={classNames('sidenav__main--wrapper', { 'hide': !visible, 'show': visible })}>

        <div className='user__section'>
          <img src='' alt='' className='user__avatar' />
          <h1 className='user__name'>{user.username}</h1>
          <h2 className='user__jobtitle'>Asistente Administrativo</h2>
          <h3 className='user__rol'>Webmaster</h3>
        </div>

        <div className='modules__section'>
          { modules ? (
            modules.map((m, i) => (
              <SideNavModule
                key={m.name}
                module={m}
              />
            ))
          ) : (null)}
          {
            <SideNavModule
              module={{
                hasSubmodules: false,
                name: 'Cerrar SesiÃ³n',
                route: '/login',
                icon: 'AiOutlineLogout',
              }}
              handleAction={handleLogout}
            />
          }
        </div>
      </nav>
      <div className={classNames('main__content', { 'content__expanded': !visible })}>
        {children}
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.auth.user,
});

const mapDispatchToProps = {
  logoutRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);
