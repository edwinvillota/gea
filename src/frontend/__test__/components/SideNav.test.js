import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import SideNav from '../../components/SideNav';
import ProviderMock from '../../__mocks__/providerMock';

describe('<SideNav />', () => {
  test('Sidenav have main wrapper', () => {
    const sidenav = mount(
      <ProviderMock>
        <SideNav />
      </ProviderMock>
    );
    expect(sidenav.find('.sidenav__main--wrapper')).toHaveLength(1);
  });

  test('Sidenav snapshot', () => {
    const sidenav = create(
      <ProviderMock>
        <SideNav />
      </ProviderMock>
    );
    expect(sidenav.toJSON()).toMatchSnapshot();
  });
});
