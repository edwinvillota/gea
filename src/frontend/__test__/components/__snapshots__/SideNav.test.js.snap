// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav /> Sidenav snapshot 1`] = `
<div
  className="main__container"
>
  <div
    className="hamburguer__control--wrapper"
  >
    <label
      className="hamburguer__label"
      htmlFor="h__check"
    >
      <svg
        className="hamburguer__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
        />
      </svg>
    </label>
    <input
      id="h__check"
      name="hamburguer"
      onClick={[Function]}
      type="checkbox"
    />
  </div>
  <nav
    className="sidenav__main--wrapper hide"
  >
    <div
      className="user__section"
    >
      <img
        alt=""
        className="user__avatar"
        src=""
      />
      <h1
        className="user__name"
      />
      <h2
        className="user__jobtitle"
      >
        Asistente Administrativo
      </h2>
      <h3
        className="user__rol"
      >
        Webmaster
      </h3>
    </div>
    <div
      className="modules__section"
    >
      <a
        className="module__wrapper"
        href="/login"
        onClick={[Function]}
      >
        <div
          className="icon__wrapper"
        >
          <svg
            className="module__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 0 1-112.7 75.9A352.8 352.8 0 0 1 512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 0 1-112.7-75.9 353.28 353.28 0 0 1-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 0 0 0-12.6z"
            />
          </svg>
        </div>
        <h3
          className="module__name"
        >
          Cerrar Sesi√≥n
        </h3>
        <div
          className="icon__wrapper"
        >
          <svg
            className="expand__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
            />
          </svg>
        </div>
      </a>
    </div>
  </nav>
  <div
    className="main__content content__expanded"
  />
</div>
`;
